xarray/tests/test_backends.py

```diff
@@ -2541,6 +2541,14 @@ def test_complex(self, invalid_netcdf, warntype, num_warns):
 
         assert recorded_num_warns == num_warns
 
+    def test_numpy_bool_(self):
+        # h5netcdf loads booleans as numpy.bool_, this type needs to be supported
+        # when writing invalid_netcdf datasets in order to support a roundtrip
+        expected = Dataset({"x": ("y", np.ones(5), {"numpy_bool": np.bool_(True)})})
+        save_kwargs = {"invalid_netcdf": True}
+        with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
+            assert_equal(expected, actual)
```

Comment:
this should be `assert_identical` to also compare attributes:
```suggestion
            assert_identical(expected, actual)
```